src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands/UpdateStoreCommand.g.cs

```diff
@@ -56,6 +57,9 @@ internal abstract class UpdateStoreCommandHandlerBase : CommandBase<UpdateStoreC
 		{
 			return null;
 		}
+		await DbContext.Entry(entity).Reference(x => x.EmailAddress).LoadAsync();
+		if(entity.EmailAddress is not null)
+			DbContext.Entry(entity.EmailAddress).State = EntityState.Deleted;
```

Comment:
the same here, not sure where we are adding the new one/update the ones to be updated

Reply:
I see now that we are doing it on the factory.... I think we need to consolidate this responsibility, somewhere, maybe in the factory, lets discuss this :)

Reply:
Implemented via IRepository as discussed